devices/HumanLogger/HumanLogger.cpp

```diff
@@ -353,6 +355,20 @@ bool HumanLogger::attach(yarp::dev::PolyDriver* poly)
         yInfo() << logPrefix << deviceName << "attach() successful";
     }
 
+    // If only one type of data is used, attachAll will not be called
+    // so we have to start the periodic thread from here
+    if(pImpl->numberOfDevicesToAttach == 1)
+    {
+        if (!pImpl->configureBufferManager()) {
+            yError() << logPrefix << "Failed to configure buffer manager for the logger.";
+        }
+
+        if(!start()) {
+            yError() << logPrefix << "Failed to start the loop.";
+            return false;
```

Comment:
This part is duplicated in the two methods, what do you think of creating a method to avoid duplication.

Reply:
You are right. I changed the modification moving all the start logic only into the `attach()` method